### PR TITLE
Enabling Staging-API Access on Website Dashboard 

### PR DESCRIPTION
**Pull request by Pavangbhat**
Date: 27-02-2024

Developer Name: Pavan Bhat

---

## Issue Ticket Number

 #v0Kn1KsLZbWWs0V9A6Pj

## Description

**This modification ensures that the root page of the website dashboard can now communicate with staging-APIs (staging-api.realdevsquad.com).**

Previously, the root page did not recognize the environment variable API_BASE_URL due to the absence of the loadEnv script. Consequently, the value of API_BASE_URL was being sourced from constants, where it was hard coded to the production API

### Documentation Updated?

- [ ] Yes
- [ ] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

</details>

### CODE REVIEW

**Comment by Pavangbhat** [comment]
closes: #676

File: helpers/loadENV.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
-window.location.hostname !== 'dashboard.realdevsquad.com'
-  ? (window.API_BASE_URL = 'https://staging-api.realdevsquad.com')
-  : (window.API_BASE_URL = 'https://api.realdevsquad.com');
+window.API_BASE_URL =
+  window.location.hostname !== 'dashboard.realdevsquad.com'
+    ? window.location.hostname !== 'localhost'
+      ? 'https://staging-api.realdevsquad.com'
+      : 'https://api.realdevsquad.com'
+    : 'https://api.realdevsquad.com';
```
Comment (Ajeyakrishna-k) [inline helpers/loadENV.js:6]: Can we please avoid using nested ternary operations?

Comment (Pavangbhat) [reply helpers/loadENV.js]: Thanks, fixed

File: constants.js [hunk 2]
```diff
@@ -1,6 +1,5 @@
-const API_BASE_URL = 'https://api.realdevsquad.com';
-const REPO_SYNC_API_URL =
-  'https://staging-sync.staging-realdevsquad-com.workers.dev';
+const API_BASE_URL = window.API_BASE_URL || 'https://api.realdevsquad.com';
+('https://staging-sync.staging-realdevsquad-com.workers.dev');
```
[the remark this answers is not in the document]
Comment (Pavangbhat) [reply constants.js:2]: Thanks, Ajey
That was miss

**Review by Ajeyakrishna-k — APPROVED** [review]

**Review by ajoykumardas12 — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Pull request merged**